### PR TITLE
[Form] document changes to form type interfaces

### PR DESCRIPTION
**Pull request by xabbuh**
| Q | A |
| --- | --- |
| Fixed tickets | #13407, #16024, #16724, #16749 |
| License | MIT |

### CODE REVIEW

File: UPGRADE-3.0.md [hunk 1]
```diff
@@ -114,6 +114,12 @@ UPGRADE FROM 2.x to 3.0
 
 ### Form
 
+ * The `getBlockPrefix()` method was added to the `FormTypeInterface` in favor of
+   the `getName()` method which has been has been removed.
```
Comment (ogizanagi) [inline UPGRADE-3.0.md]: In general, the sentence is rather as follows:
```md
The `FormTypeInterface::getName()` method has been removed in favor of 
the `FormTypeInterface::getBlockPrefix()` method.
```

Comment (stof) [inline UPGRADE-3.0.md]: @ogizanagi the BC break in 3.0 is that the method was added in the interface

File: UPGRADE-3.0.md [hunk 2]
```diff
@@ -114,6 +114,12 @@ UPGRADE FROM 2.x to 3.0
 
 ### Form
 
+ * The `getBlockPrefix()` method was added to the `FormTypeInterface` in favor of
```
Comment (stof) [inline UPGRADE-3.0.md]: ``in replacement`` would be better than ``in favor`` IMO

File: UPGRADE-3.0.md [hunk 3]
```diff
@@ -114,6 +114,12 @@ UPGRADE FROM 2.x to 3.0
 
 ### Form
 
+ * The `getBlockPrefix()` method was added to the `FormTypeInterface` in replacement of
+   the `getName()` method which has been has been removed.
+ * The `configureOptions()` method was added to the `FormTypeInterface` in replacement
```
Comment (peterrehm) [inline UPGRADE-3.0.md]: It would be good to show the different signature to reflect the change from OptionsResolverInterface to OptionsResolver

**Comment by stof** [comment]
Thank you @xabbuh.

**Pull request closed without merging**